server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/services/ActivityService.java

```diff
@@ -0,0 +1,233 @@
+                .orElseThrow(() -> new EntityNotFoundException(FISCAL_NOT_FOUND + fiscalGuid));
+        if (!projectFiscal.getProject().getProjectGuid().toString().equals(projectGuid)) {
+            throw new EntityNotFoundException(PROJECT_FISCAL + fiscalGuid + DOES_NOT_BELONG_PROJECT + projectGuid);
+        }
+
+        return activityResourceAssembler.toModel(activity);
+    }
+
+    public void deleteActivity(String projectGuid, String fiscalGuid, String activityGuid) {
+        // Get the activity
+        ActivityEntity activity = activityRepository.findById(UUID.fromString(activityGuid))
+                .orElseThrow(() -> new EntityNotFoundException(ACTIVITY_NOT_FOUND + activityGuid));
+
+        // Verify activity belongs to the specified project fiscal
+        if (!activity.getProjectPlanFiscalGuid().toString().equals(fiscalGuid)) {
+            throw new EntityNotFoundException(ACTIVITY + activityGuid + DOES_NOT_BELONG_FISCAL + fiscalGuid);
+        }
+
+        // Verify project fiscal belongs to project
+        ProjectFiscalEntity projectFiscal = projectFiscalRepository.findById(UUID.fromString(fiscalGuid))
+                .orElseThrow(() -> new EntityNotFoundException(FISCAL_NOT_FOUND + fiscalGuid));
+        if (!projectFiscal.getProject().getProjectGuid().toString().equals(projectGuid)) {
+            throw new EntityNotFoundException(PROJECT_FISCAL + fiscalGuid + DOES_NOT_BELONG_PROJECT + projectGuid);
+        }
+
+        activityRepository.deleteById(UUID.fromString(activityGuid));
+    }
+
+    public void assignAssociatedEntities(ActivityModel resource, ActivityEntity entity) {
+        if (resource.getActivityStatusCode() != null) {
+            String forestAreaCode1 = resource.getActivityStatusCode().getActivityStatusCode();
+            ActivityStatusCodeEntity activityStatusCode = loadActivityStatusCode(forestAreaCode1);
+            entity.setActivityStatusCode(activityStatusCode);
+        }
+
+        if(resource.getRiskRatingCode() != null) {
+            String riskRatingCode1 = resource.getRiskRatingCode().getRiskRatingCode();
+            RiskRatingCodeEntity riskRatingCode = loadRiskRatingCode(riskRatingCode1);
+            entity.setRiskRatingCode(riskRatingCode);
+        }
+
+        if(resource.getContractPhaseCode() != null) {
+            String contractPhaseCode1 = resource.getContractPhaseCode().getContractPhaseCode();
+            ContractPhaseCodeEntity contractPhaseCode = loadContractPhaseCode(contractPhaseCode1);
+            entity.setContractPhaseCode(contractPhaseCode);
+        }
+    }
+
+    public ActivityStatusCodeEntity loadActivityStatusCode(String activityStatusCode) {
+        return activityStatusCodeRepository
+                .findById(activityStatusCode)
+                .orElseThrow(() -> new IllegalArgumentException("ActivityStatusCode not found: " + activityStatusCode));
```

Comment:
You're calling "or else throw" but the thrown error is never handled anywhere up the chain. Are we assuming this is just a global throw that should fail the whole process?

Reply:
Catch statements added to ActivityController